bundle/jsky.app.ot/src/main/scala/jsky/app/ot/gemini/editor/targetComponent/TargetFeedback.scala

```diff
@@ -155,26 +147,62 @@ object GuidingFeedback {
   def targetAnalysis(ctx: ObsContext, mt: MagnitudeTable, target: SPTarget): List[Row] = {
     val env = ctx.getTargets
     if (target == env.getBase) baseAnalysis(ctx, mt)
-    else guideProbe(env, target).fold(List.empty[Row]) { vgp =>
+    else guideProbe(env, target).fold(List.empty[AnalysisRow]) { vgp =>
       guideStarAnalysis(ctx, mt, vgp, target).toList
     }
   }
 
   // GuidingFeedback.Rows corresponding to global errors like missing guide
   // stars.
-  def baseAnalysis(ctx: ObsContext, mt: MagnitudeTable): List[Row] =
-    AgsRegistrar.currentStrategy(ctx).fold(List.empty[Row]) { s =>
+  def baseAnalysis(ctx: ObsContext, mt: MagnitudeTable): List[AnalysisRow] =
+    AgsRegistrar.currentStrategy(ctx).fold(List.empty[AnalysisRow]) { s =>
       s.analyze(ctx, mt).filter {
         case NoGuideStarForGroup(_, _) => true
         case NoGuideStarForProbe(_, _) => true
         case _                         => false
-      }.map { a => Row(a, None, includeProbeName = true) }
+      }.map { a => AnalysisRow(a, None, includeProbeName = true) }
     }
 
   // GuidingFeedback.Row related to the given guide star itself.
-  def guideStarAnalysis(ctx: ObsContext, mt: MagnitudeTable, gp: ValidatableGuideProbe, target: SPTarget): Option[Row] =
+  def guideStarAnalysis(ctx: ObsContext, mt: MagnitudeTable, gp: ValidatableGuideProbe, target: SPTarget): Option[AnalysisRow] =
     AgsRegistrar.currentStrategy(ctx).flatMap(_.analyze(ctx, mt, gp, target.toSiderealTarget(ctx.getSchedulingBlockStart)).map { a =>
       val plo = mt(ctx, gp).flatMap(ProbeLimits(a.probeBands, ctx, _))
-      Row(a, plo, includeProbeName = false)
+      AnalysisRow(a, plo, includeProbeName = false)
     })
 }
+
+
+object BagsFeedback {
+  import BagsStatus._
+
+  private val spinner = Resources.getIcon("spinner16-transparent.png").some
+
+  sealed class BagsStatusRow(bgColor: Color, message: String, iconOpt: Option[Icon]) extends Row {
+    object feedbackLabel extends Label {
+      border = labelBorder
+      foreground = Color.DARK_GRAY
+      background = bgColor
+      text = message
+      icon = iconOpt.orNull
+      opaque = true
+      horizontalAlignment = Alignment.Left
+    }
+
+    layout(feedbackLabel) = new Constraints {
+      weightx = 1.0
+      fill = Fill.Both
+    }
+  }
+
+  case object ErrorStatusRow   extends BagsStatusRow(LIGHT_SALMON, "An error has occurred when trying to run BAGS.", None)
+  case object PendingStatusRow extends BagsStatusRow(BANANA, "Waiting for BAGS to run...", spinner)
+  case object RunningStatusRow extends BagsStatusRow(BANANA, "BAGS is running...", spinner)
+  case class  FailureStatusRow(why: String) extends BagsStatusRow(LIGHT_SALMON, s"BAGS failed: $why", None)
```

Comment:
This is more of a question for science but I would guess that if the state is idle and yet there is no AGS star for an eligible observation we should report that.  "Automatic guide star lookup found nothing." or something. Otherwise as a user I think I'd be left wondering if something is broken.